youtube_dl/extractor/funimation.py

```diff
@@ -0,0 +1,92 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .common import InfoExtractor
+from ..compat import compat_HTTPError
+from ..utils import (
+    sanitized_Request,
+    ExtractorError,
+    urlencode_postdata
+)
+
+import re
+
+class FunimationIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?funimation\.com/shows/.*/videos/official/(?P<id>[^?]+)'
+    _USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/601.2.7 (KHTML, like Gecko) Version/9.0.1 Safari/601.2.7'
+
+    _TEST = {
+        'url': 'http://www.funimation.com/shows/air/videos/official/breeze',
+        'info_dict': {
+            'id': 'AIRENG0001',
+            'title': 'Air - Breeze',
+            'ext': 'mp4',
+        },
+        'params': {
+            'format': 'lower',
+        }
+    }
+
+    def _real_initialize(self):
+        self._login()
+
+    def _login(self):
+        (username, password) = self._get_login_info()
+        self.report_login()
+        login_url = 'http://www.funimation.com/login'
+        data = urlencode_postdata({
+            'formname': 'loginForm2',
+            'email_field': username,
+            'password_field': password,
+        })
+        login_request = sanitized_Request(login_url, data)
+        login_request.add_header('Content-Type', 'application/x-www-form-urlencoded')
+
+        try:
+            login = self._download_webpage(login_request, login_url)
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
+                raise ExtractorError('%s is not available in your region.' % self.IE_NAME, expected=True)
+
```

Comment:
Alright, fixed both. Now, how could I do this squashing? I am doing everything from web expect testing locally.